support/util/imagemetadata.go

```diff
@@ -42,52 +42,26 @@ type RegistryClientImageMetadataProvider struct {
 	OpenShiftImageRegistryOverrides map[string][]string
 }
 
-// ImageMetadata returns metadata for a given image using the given pull secret
-// to authenticate. This lookup uses a cache based on the image digest. If the
-// reference of the image contains a digest (which is the mainline case for images in a release payload),
-// the digest is parsed from the image reference and then used to lookup image metadata in the
-// cache. When the image reference does not contain a digest, a lookup is made to the registry to
-// fetch the digest of the image that the tag refers to. This is because the actual image that the
-// tag is referring to could have changed. Once a digest is obtained, the cache is checked so that
-// no further fetching occurs. Only if both cache lookups fail, the image metadata is fetched and
-// stored in the cache.
+// ImageMetadata returns metadata for a given image using the given pull secret to authenticate.
+// The ICSPs/IDMSs are checked first for overrides and then the cache is checked using the image
+// pull spec. If not found in the cache, the manifest is looked up and added to the cache.
 func (r *RegistryClientImageMetadataProvider) ImageMetadata(ctx context.Context, imageRef string, pullSecret []byte) (*dockerv1client.DockerImageConfig, error) {
 
-	var (
-		repo           distribution.Repository
-		ref            *reference.DockerImageReference
-		parsedImageRef reference.DockerImageReference
-		err            error
-	)
-
-	parsedImageRef, err = reference.Parse(imageRef)
+	parsedImageRef, err := reference.Parse(imageRef)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse image reference %q: %w", imageRef, err)
 	}
 
-	// There are no ICSPs/IDMSs to process.
-	// That means the image reference should be pulled from the external registry
-	if len(r.OpenShiftImageRegistryOverrides) == 0 {
-		// If the image reference contains a digest, immediately look it up in the cache
-		if parsedImageRef.ID != "" {
-			if imageConfigObject, exists := imageMetadataCache.Get(parsedImageRef.ID); exists {
-				return imageConfigObject.(*dockerv1client.DockerImageConfig), nil
-			}
-		}
-		ref = &parsedImageRef
-	}
-
 	// Get the image repo info based the source/mirrors in the ICSPs/IDMSs
-	ref = seekOverride(ctx, r.OpenShiftImageRegistryOverrides, parsedImageRef)
+	ref := seekOverride(ctx, r.OpenShiftImageRegistryOverrides, parsedImageRef)
+	refPullSpec := ref.String()
```

Comment:
should we check for a nil ref here?

Reply:
The code didn't validate this before and in other places where `seekOverride` is called. I would speculate that this is done  because we already throw an error if `reference.Parse(imageRef)` returns an error and this same image is returned from `seekOverride` if there is an error. If you feel strongly about adding the check here, I would be glad to add it.